utilities/shared_disk.py

```
@@ -97,48 +102,123 @@ def _write_marker(ssh_conn: VMSSHConnection, file_path: str, content: str, vm_la
-        Assumes exactly two SCSI controllers with the shared disk alone on the
-        non-boot controller. This matches the current test VM layout. If VMs
-        gain additional controllers with non-shared disks, this logic would need
-        to match by exact SCSI position instead.
+    Returns:
+        str: Device path (e.g., 0 -> "/dev/vda", 2 -> "/dev/vdc").
+
+    Raises:
+        ValueError: If index exceeds the a-z device range.
+    """
+    if idx > 25:
+        raise ValueError(f"Volume index {idx} exceeds supported device range (a-z)")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)